types/parent-package-json/parent-package-json-tests.ts

```diff
@@ -1,10 +1,13 @@
 import * as parent from 'parent-package-json';
 
-const pathToParent: string = (parent() as parent.ParentPackage).path;
-const pathToParentBool: boolean = (parent() as boolean);
-const pathToParentOfCustomPath: string = (parent('/My/Cool/Folder') as parent.ParentPackage).path;
+const hasParent = parent();
+if (hasParent !== false) {
+    const pathToParent: string = parent().path;
```

Comment:
TS type inference won't work if you call `parent` again, because it might return a different thing the second time. You would have to use `hasParent` here.

Reply:
I've simplified the tests now and ds-lint check is passing. Since I am unable to invoke `parent()` multiple times I have had to remove some of the tests of the constructor.